### PR TITLE
docs: add release notes for 0.27.0

### PR DESCRIPTION
**Pull request by julian-determined-ai**
Add release notes for 0.27.0.

### CODE REVIEW

**Comment by julian-determined-ai** [comment]
the release notes did not update correctly this is not ready for review yet.

File: docs/release-notes.rst [hunk 1]
```diff
@@ -6,6 +6,28 @@
...
+
+**Release Date:** January 09, 2024
+
+**Breaking Changes**
+
+-  Experiments: Allow empty model definitions when creating experiments.
+
+-  CLI: Optional flags must come before or after positional arguments when creating experiments,
+   orderings such as the following are no longer supported ``det e create const.yaml -f .``, instead
+   use ``det e create -f const.yaml .`` or ``det e create const.yaml . -f``.
```
Comment (dzhu) [inline docs/release-notes.rst:24]:
```suggestion
-  CLI: Optional flags must come before or after positional arguments when creating experiments;
   orderings such as ``det e create const.yaml -f .`` are no longer supported. Instead, you should
   use ``det e create -f const.yaml .`` or ``det e create const.yaml . -f``.
```

File: docs/release-notes.rst [hunk 2]
```diff
@@ -6,6 +6,28 @@
...
+
+**Improvements**
+
+-  Allow checkpoint downloads through the server for ``checkpoint_storage`` type ``shared_fs`` and
```
Comment (dzhu) [inline docs/release-notes.rst:28]:
```suggestion
-  Allow checkpoint downloads through the server for ``checkpoint_storage`` types ``shared_fs`` and
```

File: docs/release-notes.rst [hunk 3]
```diff
@@ -6,6 +6,28 @@
...
+**************
+
+Version 0.27.0
+==============
+
+**Release Date:** January 09, 2024
+
+**Breaking Changes**
+
+-  Experiments: Allow empty model definitions when creating experiments.
```
Comment (dzhu) [inline docs/release-notes.rst:20]: This seems like maybe it shouldn't count as breaking. @corban-beaird?

**Review by dzhu — APPROVED** [review]

File: docs/release-notes.rst (hunk 3 above)
Comment (corban-beaird) [reply docs/release-notes.rst:20]: Yea, that's a fair point, the second bullet point is the breaking change, the first is the reason why the break is occuring

Comment (julian-determined-ai) [reply docs/release-notes.rst:20]: @corban-beaird should the first bullet be removed all together or moved to the `improvements` section?

**Review by vishnu2kmohan — APPROVED** [review]
Thanks @julian-determined-ai!

**Pull request merged**